checkov/cloudformation/checks/resource/base_resource_value_check.py

```diff
@@ -63,7 +63,7 @@ def scan_resource_conf(self, conf: dict[str, Any]) -> CheckResult:
                 # those, allowing inspected_keys in checks to use the same syntax.
                 # The last value shouldn't be changed, because it could be indeed a valid number
                 for i in range(0, len(match) - 1):
-                    if type(match[i]) == int:
+                    if type(match[i]) is int:
```

Comment:
```suggestion
                    if isinstance(match[i], int):
```

Reply:
this one is intentionally limited to only `int` because it would otherwise allow `bool` values too